pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +92,72 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) ensureWorkloadIdentityRBAC() ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	clusterMSIResourceId, err := m.doc.OpenShiftCluster.ClusterMsiResourceId()
+	if err != nil {
+		return nil, err
+	}
+	clusterMSI := m.doc.OpenShiftCluster.Identity.UserAssignedIdentities[clusterMSIResourceId.String()]
+	if strings.TrimSpace(clusterMSI.PrincipalID) == "" {
+		return nil, fmt.Errorf("cluster MSI principal ID '%s' is invalid for clusterMSIResourceId %s", clusterMSI.PrincipalID, clusterMSIResourceId.String())
+	}
+
+	resources := []*arm.Resource{}
+	managedRG := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	resources = append(resources, m.workloadIdentityResourceGroupRBAC(rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, clusterMSI.PrincipalID, managedRG))
```

Comment:
There is no need to assign rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole on the managed RG. This role will be provided on Cluster MIS with scope PlatformIdentities by customer.

Reply:
@rajdeepc2792 - Do you mean `rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole` will be added by customer to Cluster MSI and we don't have to perform this?
cc - @cadenmarchese @kimorris27 @niontive

Reply:
updated